src/components/Tooltip/TooltipRootClose.js

```diff
@@ -0,0 +1,82 @@
+import { PureComponent } from 'react'
+import { findDOMNode }   from 'react-dom'
+import PropTypes         from 'prop-types'
+
+class TooltipRootClose extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    onRootClose: PropTypes.func.isRequired,
+    rootCloseEnabled: PropTypes.bool
+  }
+
+  static defaultProps = {
+    rootCloseEnabled: true
+  }
+
+  componentDidMount() {
+    const { rootCloseEnabled } = this.props
+
+    if (rootCloseEnabled) {
+      this.addEventListeners()
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` is getting deprecated in react 17: https://developmentarc.gitbooks.io/react-indepth/content/life_cycle/update/component_will_receive_props.html

Reply:
Looks like the version of React we are using doesn't have the replacement method `static getDerivedStateFromProps` would need to update React first